### PR TITLE
Update readme

### LINKED ISSUE #2388: Missing techniques in the README
[This section](https://github.com/unitaryfund/mitiq?tab=readme-ov-file#error-mitigation-techniques) of the README and main landing page of the docs is missing some techniques. 

A complete list is available in the wiki: 

image

### LINKED ISSUE #2417: Broken Qiskit logo in the readme
image

https://github.com/unitaryfund/mitiq?tab=readme-ov-file#supported-frontends

### PR DESCRIPTION
**Pull request by purva-thakre**
## Description

Fixes #2417 
Fixes #2388 

- Adds missing techniques to the readme: QSE, PT, RSE and LRE
- Adds an `Under Construction` emoji for PT, RSE and LRE
- Updates the relative path for broken Qiskit logo

---

### License

- [x] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.

- [x] I added unit tests for new code.
- [x] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [x] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [x] I [updated the documentation](../blob/main/docs/CONTRIBUTING_DOCS.md) where relevant.
- [x] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,13 +120,13 @@ We refer to any programming language you can write quantum circuits in as a _fro
...
+| <a href="https://quantumai.google/cirq"><img src="https://raw.githubusercontent.com/quantumlib/Cirq/main/docs/images/Cirq_logo_color.png" alt="Cirq logo" width="65"/></a> | <a href="https://qiskit.org/"><img src="../../docs/source/img/frontend-logos/qiskit-logo.png" alt="Qiskit logo" width="85"/></a> | <a href="https://github.com/rigetti/pyquil"><img src="https://www.rigetti.com/uploads/Logos/logo-rigetti-gray.jpg" alt="Rigetti logo" width="75"/></a> | <a href="https://github.com/aws/amazon-braket-sdk-python"><img src="https://a0.awsstatic.com/libra-css/images/logos/aws_logo_smile_1200x630.png" alt="AWS logo" width="75"/></a> | <a href="https://pennylane.ai/"><img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/c2f96705efd4570e8755e829b11cc869b4c2287d/doc/_static/logo.png" alt="PennyLane logo"  width="30"/></a> | <a href="https://qibo.science/"><img src="https://raw.githubusercontent.com/qiboteam/qibo/master/doc/source/_static/qibo_logo_dark.svg" alt="Qibo logo" width="60"/></a> |
```
Comment (purva-thakre) [inline README.md:123]:
```suggestion
| <a href="https://quantumai.google/cirq"><img src="https://raw.githubusercontent.com/quantumlib/Cirq/main/docs/images/Cirq_logo_color.png" alt="Cirq logo" width="65"/></a> | <a href="https://qiskit.org/"><img src="docs/source/img/frontend-logos/qiskit-logo.png" alt="Qiskit logo" width="85"/></a> | <a href="https://github.com/rigetti/pyquil"><img src="https://www.rigetti.com/uploads/Logos/logo-rigetti-gray.jpg" alt="Rigetti logo" width="75"/></a> | <a href="https://github.com/aws/amazon-braket-sdk-python"><img src="https://a0.awsstatic.com/libra-css/images/logos/aws_logo_smile_1200x630.png" alt="AWS logo" width="75"/></a> | <a href="https://pennylane.ai/"><img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/c2f96705efd4570e8755e829b11cc869b4c2287d/doc/_static/logo.png" alt="PennyLane logo"  width="30"/></a> | <a href="https://qibo.science/"><img src="https://raw.githubusercontent.com/qiboteam/qibo/master/doc/source/_static/qibo_logo_dark.svg" alt="Qibo logo" width="60"/></a> |
```

To verify everything worked fine locally, I had to add `../../` to the relative path. 

If I view the file in the PR, the logo appears broken. Will wait to verify this in the RTD build but I wonder if `../../`  is only required locally?

**Comment by purva-thakre** [comment]
IDK how to fix the broken logo in both the md and the sphinx files at the same time. I had to add `../../` to the relative path to verify the output of `make docs` locally. 

Sphinx requires a relative path to where the logo is (screenshot from RTD).

image

But the `.md` file has the broken logo.

https://github.com/unitaryfund/mitiq/blob/9f84e0bb7c9f2b0077da186a305f6bb958bae774/README.md#supported-frontends
image

**Comment by purva-thakre** [comment]
Disregard my previous comment. I linked the png image directly without having to worry about the relative path.

File: README.md [hunk 2]
```diff
@@ -94,7 +94,20 @@ mitiq.qem_methods()
...
+| Layerwise Richardson Extrapolation <br> WIP 🚧          | Coming Soon <br> 🚧  | Coming Soon <br> 🚧 | [2402.04000](https://arxiv.org/abs/2402.04000) |
+
+
+In addition, we also have a noise tailoring technique currently available with limited functionality 🚧:
+
+
+| Noise-tailoring Technique                                 | Documentation                                                | Mitiq module                                                              | Paper Reference(s)                                                                                                                                 |
+| ----------------------------------------- | ------------------------------------------------------------ | ------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Pauli Twirling  <br> WIP 🚧            | [PT](https://mitiq.readthedocs.io/en/latest/guide/pt.html) <br> 🚧 | [`mitiq.pt`](https://github.com/unitaryfund/mitiq/tree/main/mitiq/pt) <br> 🚧 |  [1512.01098](https://arxiv.org/abs/1512.01098) |
```
Comment (natestemen) [inline README.md:108]: What do you think of using a single emoji and then having a note at the bottom of these two tables to indicate what 🚧 means? Here's a commit suggestion: https://github.com/unitaryfund/mitiq/commit/ab3f82fc1800c7db64ec625f967b7bb96f20847e

File: README.md [hunk 3]
```diff
@@ -107,13 +120,13 @@ We refer to any programming language you can write quantum circuits in as a _fro
...
+| <a href="https://quantumai.google/cirq"><img src="https://raw.githubusercontent.com/quantumlib/Cirq/main/docs/images/Cirq_logo_color.png" alt="Cirq logo" width="65"/></a> | <a href="https://qiskit.org/"><img src="https://raw.githubusercontent.com/unitaryfund/mitiq/main/docs/source/img/frontend-logos/qiskit-logo.png" alt="Qiskit logo" width="85"/></a> | <a href="https://github.com/rigetti/pyquil"><img src="https://www.rigetti.com/uploads/Logos/logo-rigetti-gray.jpg" alt="Rigetti logo" width="75"/></a> | <a href="https://github.com/aws/amazon-braket-sdk-python"><img src="https://a0.awsstatic.com/libra-css/images/logos/aws_logo_smile_1200x630.png" alt="AWS logo" width="75"/></a> | <a href="https://pennylane.ai/"><img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/c2f96705efd4570e8755e829b11cc869b4c2287d/doc/_static/logo.png" alt="PennyLane logo"  width="30"/></a> | <a href="https://qibo.science/"><img src="https://raw.githubusercontent.com/qiboteam/qibo/master/doc/source/_static/qibo_logo_dark.svg" alt="Qibo logo" width="60"/></a> |
```
Comment (natestemen) [inline README.md:123]: Gah long lines! Just to make sure, the only thing that has changed here is the qiskit image `src`, right?

**Review by natestemen — COMMENTED** [review]
Thanks for adding these! One suggestion to simplify it a bit.

File: README.md (hunk 3 above)
Comment (purva-thakre) [reply README.md:123]: Yup!

**Review by natestemen — APPROVED** [review]
LGTM! 🚢

**Pull request merged**